src/bokeh/models/scales.py

```diff
@@ -109,6 +111,26 @@ class CategoricalScale(Scale):
     def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
 
+class CompositeScale(Scale):
+    ''' Represent a composition of two scales, which useful for defining
+    sub-coordinate systems.
+
+    '''
+
+    # explicit __init__ to support Init signatures
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+    source_scale = Required(Instance(Scale), help="""
+    The source scale.
+    """)
+
+    target_scale = Required(Instance(Scale), help="""
+    The target scale.
+    """)
```

Comment:
These help strings don't add anything, really. What would be helpful for both of these is a brief concrete example to anchor the concepts, i.e. if we have a plot and a subplot, which one of those is the source and which one is the target?

Reply:
I will continue documenting sub-coordinates' related stuff in PR #13346.